### ISSUE TITLE
SafeAreaView - Invariant violation 

### ISSUE THREAD

**Issue by MayeuldP** [react-navigation/react-navigation#3888]
### Current Behavior

- What code are you running and what is happening? 

I m currently having troubles with SafeAreaView, concerning the layout for iPhoneX. 

I was using` SafeAreaView import from react-native`... it was ok but then I got troubles concerning the TabNavigation, it wasn't t working, so I imported `SafeAreaView from 'react-navigation'` but then I got this error. I don t know why, but if I put instead the `import from react-native`, the TabNavigation doesn't work well but still, a bit, the tab is still at the bottom of the screen, but the SafeAreaView seems to work since, I can see the background color of it, it s just the toolbar who dont take the needed padding... 

- Include a screenshot if it makes sense.
<img width="359" alt="capture d ecran 2018-04-03 a 16 34 29" src="https://user-images.githubusercontent.com/15033774/38255836-231baafc-375d-11e8-9faa-d0c75525ab47.png">

### Expected Behavior

- What do you expect should be happening? The TabNavigation should works and adapting well to iPhoneX's layout.
- Include a screenshot if it makes sense.
<img width="391" alt="capture d ecran 2018-03-27 a 10 44 28" src="https://user-images.githubusercontent.com/15033774/38256702-3301be6e-375f-11e8-9591-7c823da94ceb.png">

So basically, it s mainly concerning about the red screen, and why I get this error when I import SafeAreaView from react-navigation and not about how SafeAreaView works (this, I got it)....I checked all my import, deleted all the previous import from react-native, checked all the props... :/ 
Thanks for your help. :) 

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |last one
| react-native     |0.54.2
| node             |v8.10.0
| npm or yarn      |5.6.0

**Comment by MayeuldP** [comment]
[#1225](https://github.com/infinitered/ignite/issues/1225#issuecomment-362800224
)  Made the job...

**Comment by xSmoggeRx** [comment]
How did you fix it? I have the same problem.

**Comment by MayeuldP** [comment]
Upgrading react-native and import SafeAreaView from react-native and not react-navigation.
